python_modules/dagster/dagster/_core/external_execution/task.py

```diff
@@ -1,143 +1,71 @@
 import json
-import os
-import tempfile
-import time
 from abc import abstractmethod
-from contextlib import ExitStack
-from dataclasses import dataclass, field
-from threading import Event, Thread
 from typing import (
     Any,
     ContextManager,
-    Dict,
     Generic,
     Mapping,
     Optional,
-    Tuple,
-    Union,
 )
 
 from dagster_externals import (
-    DAGSTER_EXTERNALS_DEFAULT_INPUT_FILENAME,
-    DAGSTER_EXTERNALS_DEFAULT_OUTPUT_FILENAME,
     DAGSTER_EXTERNALS_ENV_KEYS,
+    ExternalExecutionContextData,
     ExternalExecutionExtras,
 )
-from typing_extensions import Literal, TypeAlias, TypeVar
+from typing_extensions import TypeVar
 
 import dagster._check as check
 from dagster import OpExecutionContext
 from dagster._core.definitions.data_version import DataVersion
 from dagster._core.definitions.events import AssetKey
 from dagster._core.external_execution.context import build_external_execution_context
 
-# (host, port)
-SocketAddress: TypeAlias = Tuple[str, int]
-InputIOYield: TypeAlias = Tuple[Optional[int], Mapping[str, str]]
-OutputIOYield: TypeAlias = InputIOYield
-
-
-@dataclass
-class ExternalTaskParams:
-    pass
-
-
-@dataclass
-class ExternalTaskIOParams:
-    env: Dict[str, str] = field(default_factory=dict)
-
-
 T_TaskParams = TypeVar("T_TaskParams")
-T_TaskIOParams = TypeVar("T_TaskIOParams", bound=ExternalTaskIOParams)
+T_TaskIOParams = TypeVar("T_TaskIOParams")
 
 
 class ExternalExecutionTask(Generic[T_TaskParams, T_TaskIOParams]):
     def __init__(
         self,
         context: OpExecutionContext,
         extras: Optional[ExternalExecutionExtras],
-        input_path: Optional[str] = None,
-        output_path: Optional[str] = None,
     ):
         self._context = context
         self._extras = extras
-        self._input_path = input_path
-        self._output_path = output_path
 
     def run(self, params: T_TaskParams) -> None:
-        with ExitStack() as stack:
-            tempdir = stack.enter_context(tempfile.TemporaryDirectory())
-            input_params = stack.enter_context(self._input_context_manager(tempdir, params))
-            output_params = stack.enter_context(self._output_context_manager(tempdir, params))
-            self._launch(self.get_base_env(), params, input_params, output_params)
+        with self._setup_io(params) as io_params:
+            self._launch(self.get_base_env(), params, io_params)
```

Comment:
Seems like every subclass should just implement run directly rather than forcing this structure

Reply:
ok